diff --git a/asv/benchmark.py b/asv/benchmark.py
index 3f5a9e7..44904da 100644
--- a/asv/benchmark.py
+++ b/asv/benchmark.py
@@ -446,7 +446,7 @@ class Benchmark(object):
         self._setup_cache = _get_first_attr(attr_sources, 'setup_cache', None)
         self.setup_cache_key = get_setup_cache_key(self._setup_cache)
         self.setup_cache_timeout = _get_first_attr([self._setup_cache], "timeout", None)
-        self.timeout = _get_first_attr(attr_sources, "timeout", 60.0)
+        self.timeout = _get_first_attr(attr_sources, "timeout", 600.0)
         self.code = get_source_code([self.func] + self._setups + [self._setup_cache])
         if sys.version_info[0] >= 3:
             code_text = self.code.encode('utf-8')
