diff --git a/asv_bench/benchmarks/omnisci/benchmarks.py b/asv_bench/benchmarks/omnisci/benchmarks.py
index b3dfd205..ab6b90ad 100644
--- a/asv_bench/benchmarks/omnisci/benchmarks.py
+++ b/asv_bench/benchmarks/omnisci/benchmarks.py
@@ -53,6 +53,7 @@ class TimeMerge:
         get_benchmark_shapes("omnisci.TimeMerge"),
         ["left"],
     ]
+    timeout=600
 
     def setup(self, shapes, how):
         self.df1 = generate_dataframe(
@@ -129,6 +130,7 @@ class TimeBinaryOpSeries:
 class TimeArithmetic:
     param_names = ["shape"]
     params = [get_benchmark_shapes("omnisci.TimeArithmetic")]
+    timeout=600
 
     def setup(self, shape):
         self.df = generate_dataframe(ASV_USE_IMPL, "int", *shape, RAND_LOW, RAND_HIGH)
@@ -157,6 +159,7 @@ class TimeSortValues:
         [1, 5],
         [False, True],
     ]
+    timeout=600
 
     def setup(self, shape, columns_number, ascending_list):
         self.df = generate_dataframe(ASV_USE_IMPL, "int", *shape, RAND_LOW, RAND_HIGH)
@@ -220,6 +223,7 @@ class TimeFillna:
         get_benchmark_shapes("omnisci.TimeFillna"),
         [None],
     ]
+    timeout=600
 
     def setup(self, value_type, shape, limit):
         self.df = gen_nan_data(ASV_USE_IMPL, *shape)
@@ -281,6 +285,7 @@ class TimeIndexing:
             "function",
         ],
     ]
+    timeout=600
 
     def setup(self, shape, indexer_type):
         self.df = generate_dataframe(ASV_USE_IMPL, "int", *shape, RAND_LOW, RAND_HIGH)
@@ -356,6 +361,7 @@ class TimeAstype:
 class TimeDescribe:
     param_names = ["shape"]
     params = [get_benchmark_shapes("omnisci.TimeDescribe")]
+    timeout=600
 
     def setup(self, shape):
         self.df = generate_dataframe(ASV_USE_IMPL, "int", *shape, RAND_LOW, RAND_HIGH)
